app/controllers/application_controller.rb

```diff
@@ -2,15 +2,24 @@ class ApplicationController < ActionController::Base
   helper_method :current_user
   helper_method :authenticate_user
 
+  rescue_from CanCan::AccessDenied do |exception|
+    redirect_to root_url, alert: "#{exception.message}"
+  end
```

Comment:
```suggestion
rescue_from CanCan::AccessDenied, with: :access_denied

# then write this private method below

def access_denied(error)
  flash[:error] = error.message
  redirect_to root_path
end
```

Reply:
Sorry, I don't understand. Why do we separate it into 2 methods. The first one already gets called on all the pages if the Access is restricted, no?

Reply:
Skip this change if you don't need this now. Just a suggestion, you can manually raise errors if you have to in the upcoming PRs.

Reply:
ok. thanks!